src/vaadin-message-input.js

```diff
@@ -65,6 +79,29 @@ class MessageInputElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     // Set initial height to one row
     textarea.setAttribute('rows', 1);
     textarea.style.minHeight = '0';
+
+    // Add enter handling for text area.
+    textarea.addEventListener('keydown', (e) => {
+      if (e.key === 'Enter' && !e.shiftKey) {
+        e.preventDefault();
+        e.stopPropagation();
+        this.submitMessage();
+      }
+    });
+  }
+
+  /**
+   * Submits the current value as an custom event named 'submit'.
+   * It also clears the text input and refocuses it for sending another message.
+   * In UI, can be triggered by pressing the submit button or pressing enter key when field is focused.
+   * It does not submit anything if text is empty.
+   */
+  submitMessage() {
+    if (this.text !== '') {
+      this.dispatchEvent(new CustomEvent('submit', { detail: { text: this.text } }));
+      this.text = '';
+    }
+    this.shadowRoot.querySelector('vaadin-text-area').focus();
```

Comment:
Should it focus the text-area or not? If I'm using this function to trigger the submit in a non-traditional way, it might be unexpected that the text-are becomes focused. Though it's hard to reason about without knowing a use case.

Reply:
It makes writing multiple messages more easily when the field is cleared and focus is added to be able to write the next thing. Like in Slack. The non-traditional might want some other way of doing it, but I don't know what these are. `_submit()` is now also protected so we don't have the same public API anymore.

Reply:
Yeah, I meant only for using the previously-public function, because the use case wasn't clear. Without the public function there's no issue. :+1: